pkg/provisioner/templates/containerd.go

```diff
@@ -65,7 +154,7 @@ func NewContainerd(env v1alpha1.Environment) *Containerd {
 	var version string
 
 	if env.Spec.ContainerRuntime.Version == "" {
-		version = "1.6.27"
+		version = "2.0.4"
```

Comment:
Do we really want to change the default in this PR? This has implications on Operator testing for example where we may not be specifying the containerd version.

Reply:
this changes won't affect the Operator as it has the Holodeck version pinned https://github.com/NVIDIA/gpu-operator/blob/main/.github/workflows/ci.yaml#L337. 
but if the container is a big change, I can leave the default version back to 1.6 or maybe 1.7.26 that is the latest of the 1.x.y series. WDYT?